### ISSUE TITLE
Stats widget occasionally shows zero stats

### ISSUE THREAD

**Issue by twstokes** [wordpress-mobile/WordPress-iOS#21426]
### Expected behavior
The Stats widget to always show the latest numbers accurately.

### Actual behavior
Sometimes the Stats widget shows zero stats. Bringing the app to the foreground (and maybe going to Stats) resolves it.

<img src="https://github.com/wordpress-mobile/WordPress-iOS/assets/2092798/bd2c1989-44e7-46c0-bd3e-db2d4ca7207e" width="330px" />

### Steps to reproduce the behavior

**Note: These steps will not deterministically reproduce the issue.**

1. Add the Stats Widget to the home screen
2. Use the app
3. Background the app
4. Observe the Stats Widget

**Comment by twstokes** [comment]
👋 Hey @kean since I know you did a deep dive recently into how we cache Stats, do you have any thoughts on why this may be happening? I didn't notice until `23.0`, but that doesn't mean much. Thanks!

**Comment by kean** [comment]
The widget isn't using the data that was moved from Core Data directly, but there may be a connection. I'll assign it to myself and look into it.

**Comment by kean** [comment]
I was debugging it and found a related issue where the app [overwrites](https://github.com/wordpress-mobile/WordPress-iOS/blob/bead40f3e82adc3c4a175ae4548056648e901fbe/WordPress/Classes/Stores/StatsWidgetsStore.swift#L40) the selected site ID every time you launch it:

```swift
@objc func initializeStatsWidgetsIfNeeded() {
    UserDefaults(suiteName: WPAppGroupName)?.setValue(AccountHelper.defaultSiteId, forKey: AppConfiguration.Widget.Stats.userDefaultsSiteIdKey)
    // ...
}
```

I'm not sure what the implications are, but the stats screen fails to update at least portions of the data because of it.

**Comment by kean** [comment]
I opened a PR with a fix for the "Today," "Week," and "All Time" widgets: https://github.com/wordpress-mobile/WordPress-iOS/pull/21429. There is potentially a bit more work needed around the site ID situation.

**RCA:** The initial change in stats was designed to replace Core Data with a disk cache to avoid changing behavior (only performance), but it changed to a memory cache in the final version. Because the cache is stored only in memory, the data is empty when you launch the app. It overwrites the previously downloaded good widget data with zeros. It is something I overlooked during development and testing.
